Library/Homebrew/formula_installer.rb

```diff
@@ -344,7 +344,15 @@ def expand_dependencies(deps)
     inherited_options = {}
 
     expanded_deps = Dependency.expand(formula, deps) do |dependent, dep|
-      options = inherited_options[dep.name] = inherited_options_for(dep)
+      additional_inherited = inherited_options_for(dep)
+      unless additional_inherited.nil?
+        inherited_options[dep.name] = if inherited_options[dep.name].nil?
+          additional_inherited
+        else
+          inherited_options[dep.name] + additional_inherited
+        end
+      end
+      options = inherited_options[dep.name]
```

Comment:
Right, but looks like `inherited_options_for` always returns at least `Options.new`, at least as currently implemented. But maybe best not to assume that implementation detail will always be true?